src/webapi/mapactionscontroller.cpp

```diff
@@ -18,20 +18,240 @@
+      break;
+    }
+
+    case web::AIRPORT: {
+      mapPixmap = getPixmapPosDistance(width, height, NavApp::getAirportPos(ident), distanceKm, QLatin1String(""), tr("Airport %1 not found").arg(ident));
+      break;
+    }
+  }
+  return mapPixmap;
+}
+
+MapPixmap MapActionsController::getPixmapPosDistance(int width, int height, atools::geo::Pos pos, float distanceKm,
+                                                 const QString& mapCommand, const QString& errorCase)
+{
+  if(verbose)
+    qDebug() << Q_FUNC_INFO << width << "x" << height << pos << "distanceKm" << distanceKm << "cmd" << mapCommand;
+
+  if(!pos.isValid())
+  {
+    if(errorCase == QLatin1String(""))
+    {
+      // Use current map position
+      pos.setLonX(static_cast<float>(NavApp::getMapWidget()->centerLongitude()));
+      pos.setLatY(static_cast<float>(NavApp::getMapWidget()->centerLatitude()));
+    }
+    else
+    {
+      qWarning() << Q_FUNC_INFO << errorCase;
+      MapPixmap mappixmap;
+      mappixmap.error = errorCase;
+      return mappixmap;
+    }
+  }
+
+  if(mapPaintWidget != nullptr)
+  {
+    // Copy all map settings
+    mapPaintWidget->copySettings(*NavApp::getMapWidget());
+
+    // Do not center world rectangle when resizing map widget
+    mapPaintWidget->setKeepWorldRect(false);
+
+    // Jump to position without zooming for sharp map
+    mapPaintWidget->showPosNotAdjusted(pos, distanceKm);
+
+    if(!mapCommand.isEmpty())
+    {
+      // Move or zoom map by command
+      if(mapCommand == QLatin1String("left"))
+        mapPaintWidget->moveLeft(Marble::Instant);
+      else if(mapCommand == QLatin1String("right"))
+        mapPaintWidget->moveRight(Marble::Instant);
+      else if(mapCommand == QLatin1String("up"))
+        mapPaintWidget->moveUp(Marble::Instant);
+      else if(mapCommand == QLatin1String("down"))
+        mapPaintWidget->moveDown(Marble::Instant);
+      else if(mapCommand == QLatin1String("in"))
+        mapPaintWidget->zoomIn(Marble::Instant);
+      else if(mapCommand == QLatin1String("out"))
+        mapPaintWidget->zoomOut(Marble::Instant);
+      else
+      {
+        qWarning() << Q_FUNC_INFO << "Invalid map command" << mapCommand;
+        return MapPixmap();
+      }
+    }
+
+    // Jump to next sharp level
+    mapPaintWidget->zoomIn(Marble::Instant);
+    mapPaintWidget->zoomOut(Marble::Instant);
+
+    MapPixmap mappixmap;
+
+    // The actual zoom distance
+    mappixmap.correctedDistanceKm = static_cast<float>(mapPaintWidget->distance());
+
+    if(mapCommand == QLatin1String("in") || mapCommand == QLatin1String("out"))
+      // Requested is equal to result when zooming
+      mappixmap.requestedDistanceKm = mappixmap.correctedDistanceKm;
+    else
+      // What was requested
+      mappixmap.requestedDistanceKm = distanceKm;
+
+    // Fill result object
+    mappixmap.pixmap = mapPaintWidget->getPixmap(width, height);
```

Comment:
#848 Hi @albar965, I've started using MapPaintWidget::getPixmap() here but as far as i can see this hasn't been merged yet. Still won't exclude that I may have introduced the issue on the #772 and #776 PR's...

Should I check or investigate further?

Cheers

Reply:
Would the merge the problem? Or reverting a change?
And there is still the issue with `getIls()`.

I can simply release it the next days. It's an alpha/development version. People can expect issues. :slightly_smiling_face: 

Alex

Reply:
> Nah, i don't think merging would solve the issue. At least I haven't run into it yet so i didn't fix anything in this PR. So maybe we shouldn't merge before we solve it, trying to avoid adding issues :)
> 
> Not retrieving ILS yet on this PR, I'm only that far for now:
> 

Happens during paint in the marble widget.

Reply:
I'll investigate

Reply:
Ok. Take your time.